pkg/controller/backup-operator/sync.go

```diff
@@ -66,25 +71,144 @@ func (b *Backup) processItem(key string) error {
 	}
 
 	eb := obj.(*api.EtcdBackup)
+
+	if eb.DeletionTimestamp != nil {
+		b.deletePeriodicBackupRunner(eb.ObjectMeta.UID)
+		err := b.removeFinalizerOfPeriodicBackup(eb)
+		if err != nil {
+			return err
+		}
+		return nil
+	}
 	// don't process the CR if it has a status since
 	// having a status means that the backup is either made or failed.
-	if eb.Status.Succeeded || len(eb.Status.Reason) != 0 {
+	if !isPeriodicBackup(&eb.Spec) &&
+		(eb.Status.Succeeded || len(eb.Status.Reason) != 0) {
 		return nil
 	}
-	bs, err := b.handleBackup(&eb.Spec)
-	// Report backup status
-	b.reportBackupStatus(bs, err, eb)
+
+	if isPeriodicBackup(&eb.Spec) && b.isChanged(eb) {
+		// Stop previous backup runner if it exists
+		b.deletePeriodicBackupRunner(eb.ObjectMeta.UID)
+
+		// Add finalizer if need
+		eb, err = b.addFinalizerOfPeriodicBackupIfNeed(eb)
+		if err != nil {
+			return err
+		}
+
+		// Run new backup runner
+		ticker := time.NewTicker(
+			time.Duration(eb.Spec.BackupPolicy.BackupIntervalInSecond) * time.Second)
+		ctx := context.Background()
+		ctx, cancel := context.WithCancel(ctx)
+		go b.periodicRunnerFunc(ctx, ticker, eb)
+
+		// Store cancel function for periodic
+		b.backupRunnerStore.Store(eb.ObjectMeta.UID, BackupRunner{eb.Spec, cancel})
+
+	} else if !isPeriodicBackup(&eb.Spec) {
+		// Perform backup
+		bs, err := b.handleBackup(nil, &eb.Spec)
+		// Report backup status
+		b.reportBackupStatus(bs, err, eb)
+	}
 	return err
 }
 
+func (b *Backup) isChanged(eb *api.EtcdBackup) bool {
+	backupRunner, exists := b.backupRunnerStore.Load(eb.ObjectMeta.UID)
+	if !exists {
+		return true
+	}
+	return !reflect.DeepEqual(eb.Spec, backupRunner.(BackupRunner).spec)
+}
+
+func (b *Backup) deletePeriodicBackupRunner(uid types.UID) bool {
+	backupRunner, exists := b.backupRunnerStore.Load(uid)
+	if exists {
+		backupRunner.(BackupRunner).cancelFunc()
+		b.backupRunnerStore.Delete(uid)
+		return true
+	}
+	return false
+}
+
+func (b *Backup) addFinalizerOfPeriodicBackupIfNeed(eb *api.EtcdBackup) (*api.EtcdBackup, error) {
+	ebNew := eb.DeepCopyObject()
+	metadata, err := meta.Accessor(ebNew)
+	if err != nil {
+		return eb, err
+	}
+	if !slice.ContainsString(metadata.GetFinalizers(), "backup-operator-periodic", nil) {
+		metadata.SetFinalizers(append(metadata.GetFinalizers(), "backup-operator-periodic"))
+		_, err := b.backupCRCli.EtcdV1beta2().EtcdBackups(b.namespace).Update(ebNew.(*api.EtcdBackup))
+		if err != nil {
+			return eb, err
+		}
+	} else {
+		return eb, nil
+	}
+	return ebNew.(*api.EtcdBackup), nil
+}
```

Comment:
`return ebNew.(*api.EtcdBackup), nil` will only be called  if !slice.ContainsString(metadata.GetFinalizers(), "backup-operator-periodic", nil) 

So we can simplify, unless this is not what you intended.

```
       if !slice.ContainsString(metadata.GetFinalizers(), "backup-operator-periodic", nil) {
		metadata.SetFinalizers(append(metadata.GetFinalizers(), "backup-operator-periodic"))
		_, err := b.backupCRCli.EtcdV1beta2().EtcdBackups(b.namespace).Update(ebNew.(*api.EtcdBackup))
		if err != nil {
			return eb, err
		}
                return ebNew.(*api.EtcdBackup), nil
	}
            return eb, nil
	
}
```